tables/vpc_flow_log/vpc_flow_log_table.go

```diff
@@ -54,56 +48,64 @@ func (c *VpcFlowLogTable) EnrichRow(row *VpcFlowLog, sourceEnrichmentFields sche
 
 	// Record standardization
 	row.TpID = xid.New().String()
-
-	// TODO is source type actually the source, i.e compressed file source etc>
-	// should these all be filled in by the source???
-	row.TpSourceType = c.Identifier()
-	//row.TpSourceName = ???
-	//row.TpSourceLocation = ???
 	row.TpIngestTimestamp = time.Now()
 
-	// Hive fields
-	if row.AccountID != nil {
-		row.TpIndex = *row.AccountID
-	} else {
-		row.TpIndex = "default"
-	}
-
-	// TODO: How to handle if the log don't have timestamp value
-	// populate the year, month, day from start time
-	if row.Timestamp != nil {
-		// convert to date in format yy-mm-dd
-		row.TpDate = row.Timestamp.In(time.UTC).Truncate(24 * time.Hour)
-		row.TpTimestamp = *row.Timestamp
-	} else if row.Start != nil {
-		// convert to date in format yy-mm-dd
-		// TODO is Start unix millis?? if so why do we convert it for TpTimestamp
-		row.TpDate = time.UnixMilli(*row.Start).Truncate(24 * time.Hour)
-
-		//convert from unis seconds to milliseconds
-		row.TpTimestamp = time.Unix(0, int64(*row.Start*1000)*int64(time.Millisecond))
-	} else {
-		// TODO: is it correct to fallback to the current time
-		t := time.Now()
-		row.TpDate = time.UnixMilli(t.UnixMilli()).Truncate(24 * time.Hour)
-		row.TpTimestamp = t
-	}
-
-	//row.TpAkas = ???
-	//row.TpTags = ???
-	//row.TpDomains = ???
-	//row.TpEmails = ???
-	//row.TpUsernames = ???
-
 	// ips
 	if row.SrcAddr != nil {
 		row.TpSourceIP = row.SrcAddr
 		row.TpIps = append(row.TpIps, *row.SrcAddr)
 	}
+	if row.PktSrcAddr != nil {
+		row.TpIps = append(row.TpIps, *row.PktSrcAddr)
+	}
 	if row.DstAddr != nil {
 		row.TpDestinationIP = row.DstAddr
 		row.TpIps = append(row.TpIps, *row.DstAddr)
 	}
+	if row.PktDstAddr != nil {
+		row.TpIps = append(row.TpIps, *row.PktDstAddr)
+	}
+
+	// TpIndex
+	if row.InterfaceID != nil {
+		row.TpIndex = *row.InterfaceID
+	} else if row.VPCID != nil {
+		row.TpIndex = *row.VPCID
+	} else if row.SubnetID != nil {
+		row.TpIndex = *row.SubnetID
+	} else {
+		row.TpIndex = "default"
+	}
```

Comment:
@cbruno10 thoughts on this for `tp_index`?

Personally, I think it should be the `VPC ID` or `default`.

Reply:
@graza-io I like the idea of having `tp_index` be the main resource that the logs are about/the log generator. This allows users to construct more specific queries when querying, e.g., `select * from aws_vpc_flow_log where tp_index = 'subnet-12345';` to find all flow logs related to a particular subnet, taking advantage of the partitioning.

Reply:
Based on above code it's not clear what my tp_index is as it can be one of 3 fields or a default...

Which means if I did a `where tp_index = 'subnet-12345'` I wouldn't get all my records related to that subnet if any of them had `tp_index` populated by an interface or vpc ID.